aopy/data/base.py

```diff
@@ -159,17 +159,101 @@ def load_preproc_lfp_data(preproc_dir, subject, te_id, date, cached=True):
         subject (str): Subject name
         te_id (int): Block number of Task entry object 
         date (str): Date of recording
+        drive_number (int): drive number for multiple recordings. 1-based indexing.
 
     Returns:
-        dict: lfp data
+        ndarray: numpy array of lfp data from hdf
         dict: Dictionary of lfp metadata
     '''
     filename = get_preprocessed_filename(subject, te_id, date, 'lfp')
     preproc_dir = os.path.join(preproc_dir, subject)
-    data = load_hdf_data(preproc_dir, filename, 'lfp_data', cached=cached)
-    metadata = load_hdf_group(preproc_dir, filename, 'lfp_metadata', cached=cached)
+    
+    group_names = list_root_groups(preproc_dir, filename)
+    num_drives = len(group_names)
+        
+    if drive_number:
+        data = load_hdf_data(preproc_dir, filename, f'drive{drive_number}/lfp_data', cached=cached)
+        metadata = load_hdf_group(preproc_dir, filename, f'drive{drive_number}/lfp_metadata', cached=cached)
+    else:
+        if num_drives > 1:
```

Comment:
shouldn't there be 2 group names for a file with lfp_data and lfp_metadata? i'm confused how this would work. it might be better to explicitly check if the group names contain 'drive1' or 'lfp_data' (or 'ap_data', etc. for other fns)